src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState_Checksum.cs

```diff
@@ -158,30 +156,27 @@ public async Task<Checksum> GetChecksumAsync(ProjectId projectId, CancellationTo
         {
             throw ExceptionUtilities.Unreachable();
         }
+    }
 
-        static Checksum GetVersionMapChecksum(SolutionCompilationState @this)
-        {
-            // We want the projects in sorted order so we can generate the checksum for the
-            // source-generation-execution-map consistently.
-            var sortedProjectIds = SolutionState.GetOrCreateSortedProjectIds(@this.SolutionState.ProjectIds);
-            var supportedCount = sortedProjectIds.Count(
-                static (projectId, @this) => RemoteSupportedLanguages.IsSupported(@this.SolutionState.GetRequiredProjectState(projectId).Language),
-                @this);
-
-            // For each project, we'll add one checksum for the project id and one for the version map.
-            using var _ = ArrayBuilder<Checksum>.GetInstance(2 * supportedCount, out var checksums);
+    public SourceGeneratorExecutionVersionMap GetFilteredSourceGenerationExecutionMap(ProjectCone? projectCone)
+    {
+        var builder = this.SourceGeneratorExecutionVersionMap.Map.ToBuilder();
 
-            foreach (var projectId in sortedProjectIds)
+        foreach (var (projectId, projectState) in this.SolutionState.ProjectStates)
+        {
+            if (!RemoteSupportedLanguages.IsSupported(projectState.Language))
             {
-                var projectState = @this.SolutionState.GetRequiredProjectState(projectId);
-                if (!RemoteSupportedLanguages.IsSupported(projectState.Language))
-                    continue;
-
-                checksums.Add(projectId.Checksum);
-                checksums.Add(Checksum.Create(@this.SourceGeneratorExecutionVersionMap[projectId], static (v, w) => v.WriteTo(w)));
+                builder.Remove(projectId);
+            }
+            else if (projectCone != null && !projectCone.Contains(projectId))
```

Comment:
>else if (projectCone != null && !projectCone.Contains(projectId)) [](http://example.com/codeflow?start=12&length=65)

a projectCone doesn't usually span multiple projects right? I'm just wondering whether it would be better to build this up by testing for inclusion rather than reduce this by testing for exclusion.

Reply:
a project-cone very often spans multiple projects.  it represents a project and all the projects that project *depends* on.

Reply:
> I'm just wondering whether it would be better to build this up by testing for inclusion rather than reduce this by testing for exclusion.

that would work as well.  I don't have strong feelings.

Reply:
That was a great explanation, thank you! Can you put a blurb in the code so I don't have that thought again?

Reply:
Will do.